components/elasticsearch/scripts/create.py

```diff
@@ -262,14 +267,96 @@ def main():
                         'index already exists...')
 
         if http_request('http://{0}:{1}/cloudify_storage'.format(
-                es_endpoint_ip, es_endpoint_port), method='HEAD'):
+                es_endpoint_ip, es_endpoint_port), method='HEAD').code == 200:
             utils.error_exit('\'cloudify_storage\' index already exists on '
                              '{0}, terminating bootstrap...'.format(
                                  es_endpoint_ip))
         _configure_elasticsearch(host=es_endpoint_ip, port=es_endpoint_port)
 
+    if utils.is_upgrade:
+        restore_upgrade_data(es_endpoint_ip, es_endpoint_port)
+
+    if not es_endpoint_port:
+        utils.systemd.stop(ES_SERVICE_NAME, append_prefix=False)
+
     ctx.instance.runtime_properties['es_endpoint_ip'] = es_endpoint_ip
 
 
-if utils.is_install:
-    main()
+def _get_es_install_port():
+    es_props = utils.ctx_factory.get(ES_SERVICE_NAME, upgrade_props=False)
+    return es_props['es_endpoint_port']
+
+
+def _get_es_install_endpoint():
+    es_props = utils.ctx_factory.get(ES_SERVICE_NAME, upgrade_props=False)
+    if es_props['es_endpoint_ip']:
+        es_endpoint = es_props['es_endpoint_ip']
+    else:
+        es_endpoint = ctx.instance.host_ip
+    return es_endpoint
+
+
+def dump_upgrade_data():
+
+    if os.path.exists(DUMP_SUCCESS_FLAG):
+        return
+
+    endpoint = _get_es_install_endpoint()
+    port = _get_es_install_port()
+    storage_endpoint = 'http://{0}:{1}/cloudify_storage'.format(endpoint,
+                                                                port)
+    types = ['provider_context', 'snapshot']
+    ctx.logger.info('Dumping upgrade data: {0}'.format(types))
+    type_values = []
+    for _type in types:
+        res = http_request('{0}/_search?q=_type:{1}&size=10000'
+                           .format(storage_endpoint, _type),
+                           method='GET')
+        if not res.code == 200:
+            raise RuntimeError('Failed fetching type {0} from cloudify_storage'
```

Comment:
ctx.abort_operation(message).
Any raise should be replaced with it to terminate the entire operation's execution